frontend/packages/policy-editor/src/utils/index.ts

```diff
@@ -88,13 +56,13 @@ export const mapPolicyRulesBackendObjectToPolicyRuleCard = (
       resource.map((resource) => mapResourceFromBackendToResource(resource)),
     );
 
-    const subjectTitles = mapPolicySubjectToSubjectTitle(subjectOptions, r.subject);
+    const subjectIds: string[] = extractPolicyIdsFromPolicySubjects(r.subject);
 
     return {
       ruleId: id,
       actions: r.actions,
       description: r.description,
-      subject: subjectTitles,
+      subject: subjectIds,
```

Comment:
Any reason for the type being singular and the value being plural? 🫣

Reply:
I have considered the same and decided not to change this in the current pull request, as the alteration also affects the backend. I see there is room for improvement in this part of the Studio. Additionally, the API doesn't seem optimal for the frontend, leading to unnecessary data mapping in various places. Here are two alternatives to address this:

1. Develop a more better API interface for the frontend to simplify the logic.
2. Perform data mapping immediately after retrieving the data from the backend and create internal models that are more self-explanatory. Then, map them back to the expected format just before sending the data back to the backend.


This is a bigger task than the fix itself in this PR. We need to merge this PR into the main code before we can deploy it, so I've chosen to limit the changes to the minimum by only fixing the bug itself. Did it make sense? 😊